backend/server/controllers/checklists.js

```diff
@@ -81,16 +87,42 @@ module.exports = {
       }
       // No validation is done to prevent creating a checklist for a week that doesn't exist.
       // Arguably, this is a feature, since the number of weeks can change.
-      const result = await Checklist.create({
+      let result = await Checklist.findOrCreate({where: {
         week: req.body.week,
-        courseName: 'doot',
-        list: req.body.checklist,
-        courseInstanceId: req.body.courseInstanceId,
-        maxPoints: req.body.maxPoints
-      })
+        courseInstanceId: req.body.courseInstanceId
+      }})
+      //Update maxPoints. This cannot be done with findOrCreate as by default courses have null as maxPoints
```

Comment:
Why is it not possible? Can't findOrCreate supply some other value as `defaults`?

Reply:
findOrCreate tries to find a course where all given fields match. Frontend seems to always send 0 for value of maxPoints (maybe a bug?) if its not defined and that would cause findOrCreate to create a new checklist.

Correct approach would be using checklistId here, but I wanted to avoid modifying frontend for this change.

Reply:
Yes, but the `defaults` value supplied for `findOrCreate` does not affect the result when finding a row (like `where`), i.e. they're not considered as conditions when finding a row, but they are included when creating a row.

Reply:
But that would not update maxPoints if its already set?

Reply:
That is a good point, I take this point back.